File: packages/cli/src/commands/react.ts
import { Command } from "commander";
import { sign } from "@chat-mcp/shared";
import { loadConfig } from "../config.js";
import { ApiClient } from "../api.js";

export const reactCommand = new Command("react")
  .description("React to a message")
  .argument("<message-id>", "Message ID (or prefix)")
  .argument("<emoji>", "Emoji or shortcode")
  .action(async (messageId, emoji) => {
    const config = loadConfig();
    if (!config.ssh_key_path) {
      console.error("No SSH key configured. Run: chat auth register");
      process.exit(1);
    }

    const api = new ApiClient(config);

    const reactionPayload = {
      message_id: messageId,
      emoji,
      author_id: config.participant_id,
    };
    const signature = await sign(config.ssh_key_path, reactionPayload);

    await api.post(`/messages/${messageId}/reactions`, { emoji, signature });
    console.log(`Reacted ${emoji} to #${messageId.slice(0, 8)}`);
  });

export const unreactCommand = new Command("unreact")
  .description("Remove a reaction")
  .argument("<message-id>", "Message ID (or prefix)")
  .argument("<emoji>", "Emoji or shortcode")
  .action(async (messageId, emoji) => {
    const config = loadConfig();
    const api = new ApiClient(config);
    await api.delete(`/messages/${messageId}/reactions/${emoji}`);
    console.log(`Removed ${emoji} from #${messageId.slice(0, 8)}`);
  });

File: src/chat_mcp/cli/commands/react.py
"""Reaction commands."""

from __future__ import annotations

import asyncio

import click


@click.command()
@click.argument("message_id")
@click.argument("emoji")
def react(message_id: str, emoji: str):
    """Add a reaction to a message."""
    from chat_mcp.cli.client import chat_client

    async def run():
        await chat_client.post(f"/messages/{message_id}/reactions", json={"emoji": emoji})
        click.echo(f"Reacted {emoji} to {message_id[:8]}")
        await chat_client.close()

    asyncio.run(run())


@click.command()
@click.argument("message_id")
@click.argument("emoji")
def unreact(message_id: str, emoji: str):
    """Remove a reaction from a message."""
    from chat_mcp.cli.client import chat_client

    async def run():
        await chat_client.delete(f"/messages/{message_id}/reactions/{emoji}")
        click.echo(f"Removed {emoji} from {message_id[:8]}")
        await chat_client.close()

    asyncio.run(run())
